Add 3 files to export

TheRealPaco opened the pull request:
I am writting a packge for [eosio signing request (esr) ](https://github.com/eosio-eps/EEPs/blob/master/EEPS/eep-7.md#ESR---The--EOSIO-Signing-Request--protocol) and I need to import the following files (serialize.dart, eosdart_base.dart, transaction.dart) to create the signing request model and serialize and deserialize the request. Would you mind adding those 3 exports to the package. it would be better for me then using a fork repository like I do for now.

Thanks

TheRealPaco committed:
add import

TheRealPaco committed:
Merge pull request #1 from TheRealPaco/feature/update-export-for-dart-esr

export for dart-esr

TheRealPaco committed:
Update serialize.dart

Remove print

TheRealPaco committed:
Remove client, examples and test

TheRealPaco committed:
Merge pull request #2 from TheRealPaco/bugfix/client.dart-causes-errors-with-flutter-web

Remove client, examples and test

TheRealPaco committed:
Add getAccountKey state path

TheRealPaco committed:
Merge branch 'master' into feature/update-getKeyAccount-to-new-endpoint

TheRealPaco committed:
Merge pull request #4 from TheRealPaco/feature/update-getKeyAccount-to-new-endpoint

Feature/update get key account to new endpoint

TheRealPaco committed:
add numeric to exports

TheRealPaco committed:
Merge pull request #5 from TheRealPaco/feature/expose-numeric

add numeric to exports

fzxu commented on lib/src/client.dart:
```diff
@@ -351,6 +351,21 @@ class EOSClient {
+        body: {"public_key": pubKey});
+
+    if (response.statusCode >= 300) {
+      throw response.body;
```
can we do better instead of throw the body?

TheRealPaco replied:
Finnaly, I don't need this anymore so I'll remove it from the pull request

TheRealPaco committed:
Change eosdart_ecc to therealpaco fork

TheRealPaco committed:
Merge pull request #6 from TheRealPaco/feature/change-eosdart-ecc-package

Change eosdart_ecc to therealpaco fork

fzxu commented:
any reason to ref to the fork?

TheRealPaco commented:
> any reason to ref to the fork?

Sorry for not responding earlier, 
I am working on a port for greymass anchor-link and I need to recover an eos public key from some signed data, so I added the feature to eosdart_ ecc and dart told me me that the eosdart package and my dart-anchor-link package need to reference the same eosdart_ecc. So for testing, I just changed it to use my fork for simplicity's sake and it updated the pull request which I had kind of forgot about. I will create a pull request for eosDart_ecc and I will change it back. After that I will change it to use the official eosDart_ecc package and also update eosDart with your other comment so we can look into merging it.

TheRealPaco committed:
Revert "Add getAccountKey state path"

This reverts commit 117d8d0e1316b0c369914c32c712c22c25611183.

fzxu commented:
@TheRealPaco I've merged that change and release 0.4.3

7flash commented:
Hello @TheRealPaco :)
I am working on dartesr package that has similar purpose, do you want to collaborate?

https://github.com/digital-scarcity/dartesr

TheRealPaco committed:
update eosdart_ecc version

TheRealPaco committed:
Fix serializing error if the value was null

TheRealPaco commented on lib/src/serialize.dart:
```diff
@@ -598,9 +598,9 @@ void serializeStruct(Type self, SerialBuffer buffer, Object data,
+  Map dy = data as dynamic;
   for (var field in self.fields) {
-    if (dy[field.name] != null) {
+    if (dy.containsKey(field.name)) {
```
In eosjs serialize `if (field.name in data)` check if the map contain the field not if the field is not null. Deserializing a null value work fine but this was causing an error when serializing null value.

TheRealPaco commented:
> Hello @TheRealPaco :)
> I am working on dartesr package that has similar purpose, do you want to collaborate?
> 
> https://github.com/digital-scarcity/dartesr

Hi, 
Yes we can but this is my last week full time on this as I was working as an intern for EOS Nation and I am going back to university. I might do some hours here and there during my session depending on my availability.

For [dart-esr](https://github.com/EOS-Nation/dart-esr), the master branch work fine but I have redone it to be more lke greymass eosio signing request js version (for easier integration with the anchor link port to dart). The package is working and almost complete for a v1. Right now, it's more finalizing things, cleaning the code, adding tests, and add documentation. There is example though if you want to take a look at the up top date version [https://github.com/EOS-Nation/dart-esr/tree/feature/complete-port-of-esr-js](https://github.com/EOS-Nation/dart-esr/tree/feature/complete-port-of-esr-js). 

Also, I'll probably be merging the branch soon and pushing it to pub.dev.

fzxu commented:
@TheRealPaco let me know once the PR is ready for review.

TheRealPaco commented:
> @TheRealPaco let me know once the PR is ready for review.

It is ready. finally I do not need many changes, just exposing those 4 files and fixing the serializing bug with null value. If I have more things in the future,I'll do another PR.
